### PR TITLE
Added useFeatureFlagWithPayload for react native

### PR DESCRIPTION
**Pull request by Jonatthu**
Enabling feature flags with payload, this should be essential to have...

## Problem

Enables feature flags with payload for react native

## Changes

It just exposes a new hook for this.

### Bump level
- [ ] Major
- [x] Minor
- [ ] Patch

### Libraries affected
- [ ] All of them
- [ ] posthog-web
- [ ] posthog-node
- [x] posthog-react-native

### Changelog notes

- Added support for feature flags payload for react native useFeatureFlagWithPayload

### CODE REVIEW

File: posthog-react-native/src/hooks/useFeatureFlag.ts [hunk 1]
```diff
@@ -18,3 +19,26 @@ export function useFeatureFlag(flag: string): string | boolean | undefined {
 
   return featureFlag
 }
+
+export function useFeatureFlagWithPayload(flag: string): [string | boolean | undefined, JsonType | undefined]   {
+  const posthog = usePostHog()
+
+  const [featureFlag, setFeatureFlag] = useState<boolean | string | undefined>(posthog?.getFeatureFlag(flag))
+  const [featureFlagPayload, setFeatureFlagPayload] = useState<JsonType | undefined>(posthog?.getFeatureFlag(flag))
```
Comment (neilkakkar) [inline posthog-react-native/src/hooks/useFeatureFlag.ts:27]:
```suggestion
  const [featureFlagPayload, setFeatureFlagPayload] = useState<JsonType | undefined>(posthog?.getFeatureFlagPayload(flag))
```

Comment (benjackwhite) [inline posthog-react-native/src/hooks/useFeatureFlag.ts:27]: Not sure I love that these two can get out of sync. 99% of the time this shouldn't happen, but still. I think we should have one state object for both values like:
```suggestion
  const [featureflagState, setFeatureFlagState] = useState<[string | boolean | undefined, JsonType | undefined]>
```

meaning later we can just do 
```
setFeatureFlag([
  posthog.getFeatureFlag(flag)),
  posthog.getFeatureFlagPayload(flag))
])
```

**Review by benjackwhite — CHANGES_REQUESTED** [review]
This is a great addition! One small tweak requested as I feel its important we get the lifecycle right. If you don't have time, let me know and I can create a PR to go in that direction

File: posthog-react-native/src/hooks/useFeatureFlag.ts (hunk 1 above)
Comment (Jonatthu) [reply posthog-react-native/src/hooks/useFeatureFlag.ts]: Made the change! I did not sleep enough when I made this haha

**Comment by Jonatthu** [comment]
@benjackwhite I think this is ready to go!

**Comment by Jonatthu** [comment]
@benjackwhite @neilkakkar I have added sentry integration, it has been tested.

image
image

**Review by marandaneto — CHANGES_REQUESTED** [review]
@Jonatthu can you split the PR into 2 PRs, one for the `useFeatureFlagWithPayload` and the other one for the Sentry integration? it's pretty much a copy and paste.
The idea is that if something goes wrong, we can easily revert, it's also easier to read the history of changes.
Thanks for doing this.

**Comment by Jonatthu** [comment]
@marandaneto Done

**Review by marandaneto — COMMENTED** [review]
> @marandaneto Done

thanks, feel free to open a new PR with the Sentry bits.

**Review by marandaneto — APPROVED** [review]
LGTM but let's wait @benjackwhite to double-check it since he requested changes.

**Comment by marandaneto** [comment]
@Jonatthu please add an entry to the [changelog](https://github.com/PostHog/posthog-js-lite/blob/master/posthog-react-native/CHANGELOG.md) so we can get it merged, thanks for the PR.

**Comment by Jonatthu** [comment]
@marandaneto Done, Can we merge it now?

**Pull request merged**